feature_engine/selection/drop_psi_features.py

```diff
@@ -0,0 +1,537 @@
+            pd.DataFrame(basis_binned)
+            .merge(
+                pd.DataFrame(meas_binned),
+                right_index=True,
+                left_index=True,
+                how="outer",
+            )
+            .fillna(0)
+        )
+        binning.columns = ["basis", "meas"]
+
+        # Compute the PSI value.
+        psi_value = self._psi(binning.basis.to_numpy(), binning.meas.to_numpy())
+
+        return psi_value
+
+    def _psi(self, d_basis, d_meas):
+        """
+        Compute the PSI value
+
+        Parameters
+        ----------
+        d_basis : np.array.
+            Proxy for the basis distribution.
+
+        d_meas: np.array.
+            Proxy for the measurement distribution.
+
+        Returns
+        -------
+        psi_value: float.
+            PSI value.
+        """
+        # Calculate the ratio of samples in each bin
+        basis_ratio = d_basis / d_basis.sum()
+        meas_ratio = d_meas / d_meas.sum()
+
+        # Necessary to avoid divide by zero and ln(0). Has minor impact on PSI value.
+        basis_ratio = np.where(
+            basis_ratio <= 0, self.min_pct_empty_buckets, basis_ratio
+        )
+        meas_ratio = np.where(meas_ratio <= 0, self.min_pct_empty_buckets, meas_ratio)
+
+        # Calculate the PSI value
+        psi_value = np.sum(
+            (meas_ratio - basis_ratio) * np.log(meas_ratio / basis_ratio)
+        )
+
+        return psi_value
+
+    def _split_dataframe(self, X):
+        """
+        Split a dataframe according to a cut-off value and return two dataframes:
+        one with values above the cut-off and one with values below or equal
+        to the cut-off.
+        The cut-off value is associated to a specific column.
+
+        The cut-off can be defined in two ways:
+
+            - Considering all observations. In that case a split fraction of 0.25
+            will ensure 25% of the observations are in the "below cut-off"
+            dataframe. The final number of observations in the "below cut-off" may
+            be slightly different than 25% as all observations with the same value
+            are put in the same split.
+
+            - Considering the values of the observations. The cut-off applies to
+            the distinct values. In that case a split fraction of 0.25 will ensure
+            25% of the distinct values are in the "below cut-off" dataframe;
+            regardless of the number of observations.
+
+        Parameters
+        ----------
+        X : pandas dataframe
+            The original dataset. Correspond either to the measurement dataset
+            or will be spit into a measurement and a basis dataframe.
+
+        Returns
+        -------
+        pandas dataframe with value below the
+        basis pandas dataframe
+        """
+        # Identify the values according to which the split must be done.
+        if self.split_col == "use_df_index":
+            reference = pd.Series(X.index.to_list()).dropna()
```

Comment:
can the index have na?

Reply:
Yes, technically it can. I can even have multiple NA's
![image](https://user-images.githubusercontent.com/32060943/136450039-77df46e4-07ff-4ae2-9a0d-c9cfd9696cb5.png)

Reply:
Nice. I wasn't sure about this